service/src/test/java/uk/gov/hmcts/reform/fpl/service/LocalAuthorityValidationServiceTest.java

```diff
@@ -0,0 +1,100 @@
+import uk.gov.hmcts.reform.fpl.request.RequestData;
+import uk.gov.hmcts.reform.rd.model.Organisation;
+
+import java.util.List;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {LocalAuthorityValidationService.class, OrganisationService.class})
+class LocalAuthorityValidationServiceTest {
+
+    @Autowired
+    private LocalAuthorityValidationService validationService;
+
+    @MockBean
+    private RequestData requestData;
+
+    @MockBean
+    private OrganisationService organisationService;
+
+    @MockBean
+    private FeatureToggleService featureToggleService;
+
+    @MockBean
+    private LocalAuthorityNameLookupConfiguration localAuthorityNameLookupConfiguration;
+
+    @MockBean
+    private LocalAuthorityService localAuthorityNameService;
+
+    private static final String LOCAL_AUTHORITY_CODE = "SA";
+    private static final String LOCAL_AUTHORITY_NAME = "Swansea City Council";
+    private static final String USER_ID = "a3850cb6-36ce-4612-b8c0-da00d57f1537";
+    private static final String ORGANISATION_IDENTIFIER = "123";
+
+
+    @BeforeEach
+    void setup() {
+        given(requestData.userId()).willReturn(USER_ID);
+        given(localAuthorityNameService.getLocalAuthorityCode()).willReturn(LOCAL_AUTHORITY_CODE);
+        given(localAuthorityNameLookupConfiguration.getLocalAuthorityName(LOCAL_AUTHORITY_CODE))
+            .willReturn(LOCAL_AUTHORITY_NAME);
+    }
+
+    @Test
+    void shouldSuccessfullyValidateWhenUserIsOnboardedAndToggleEnabled() {
+        Organisation organisation = Organisation.builder().organisationIdentifier(ORGANISATION_IDENTIFIER).build();
+        given(organisationService.findOrganisation()).willReturn(organisation);
+
+        final List<String> validationErrors = validationService.validateIfUserIsOnboarded();
+
+        assertThat(validationErrors).isEmpty();
+    }
+
+    @Test
+    void shouldNotValidateWhenUserHasNotBeenOnboarded() {
+        given(organisationService.findOrganisation()).willReturn(Organisation.builder().build());
+
+        final List<String> validationErrors = validationService.validateIfUserIsOnboarded();
+
+        assertThat(validationErrors).containsExactly(
+            "Register for an account.",
+            "You cannot start an online application until you’re fully registered.",
+            "Ask your local authority’s public law administrator for help with registration.");
+    }
+
+    @Test
+    void shouldNotValidateWhenToggleIsEnabled() {
+        given(featureToggleService.isAllowCaseCreationForUsersNotOnboardedToMOEnabled(LOCAL_AUTHORITY_NAME))
+            .willReturn(true);
+
+        validationService.validateIfUserIsOnboarded();
```

Comment:
most important thing is to validate that  validationService.validateIfUserIsOnboarded() returns empty list, which is not checked here